enhancements/cloud-integration/tokenized_auth_enablement_operators_on_cloud.md

```diff
@@ -0,0 +1,527 @@
+possible, native integration with core platform operators when clusters use tokenized cloud auth, driving the use of 
+layered products.
+
+As the Hypershift team, where the only credential mode for clusters/customers is STS (on AWS) , the Red Hat branded 
+Operators that must reach the AWS API, should be enabled to work with STS credentials in a consistent, and automated 
+fashion that allows customer to use those operators as easily as possible, driving the use of layered products.
+
+As an operator team that may already be supporting cloud credential provisioning, including a manner for allowing
+operand instances to have a unique set of credential per instance, I want my solution to keep working (Quay Operator).
+
+
+### Goals
+
+Allow Day-2 Operators to access cloud provider resources as seamlessly as possible when being installed, used and
+updated on STS enabled clusters.
+
+While providing the above goal, allow for multi-tenancy. In this sense multi-tenancy means that an operator may
+provide different cloud provider credentials per operand such that operator users may uniquely access non-shared cloud
+resources.
+
+Guide and assist Day-2 Operator admins in providing the required cloud provider credentials matched to their permission
+needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (STS always) and non-HyperShift clusters.
+
+### Non-Goals
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for STS enabled clusters (at least 
+for HyperShift)
+
+
+
+## Proposal
+
+[NB: these are all short now, just adding to them as we outline what each involved component will do in the solution]
+
+Cloud Credential Operator (CCO) changes: Add an STS mode, distinct from disabled/manual mode that will look for and
+process CredentialRequests referenced in Operator CRs. This will operate in one of two ways depending on which other parts 
+of this EP for other components are adopted: specifically HyperShift/Rosa adding CCO with STS mode or continuing with the
+current support for the pod identity webhook).
+- CCO STS mode flow will work by adding a "role-arn"-type field on operator added CredentialsRequest objects, this will 
+trigger CCO to annotate the ServiceAccount on behalf of the operator.
+- Webhook mode will work by annotating the ServiceAccount triggering the projection of the service account tokens into pods
+ created for the operator (same method and resultant actions as: https://github.com/openshift/hypershift/pull/1351)
+
+HyperShift changes: Possibly some changes to logic for Pod Identity Webhooks as needed for fine-grained credential
+management to allow for multi-tenancy (see definition earlier). Also, need to add pod identity webhooks targeting other
+cloud providers and change the annotations to more generic naming.
+
+OperatorHub changes: Allow for import of additional, well-known, ENV variables on the Subscription object. These fields
```

Comment:
Why do we need ENV vars on the subscription? I'd like to avoid "configuring" operators before install. Instead we should just install the operator and it will tell the admin or user in the CR status "hey I need xyz to move forward" - and CCO makes that xyz easier and consistent / uniform. Am I missing something?

Reply:
I think this is covered now in the EP